src/integrationTest/java/uk/gov/hmcts/darts/authentication/controller/impl/TokenValidatorTest.java

```diff
@@ -0,0 +1,150 @@
+                                       Matchers.containsString("An error occurred while attempting to decode the Jwt: Signed JWT rejected: Invalid signature")))
+            .andReturn();
+    }
+
+    @Test
+    void checkTokenExpiry() throws Exception {
+        DartsTokenGenerator token = DartsTokenGenerator.builder().useGlobalKey(true).issuer(configurationProperties.getIssuerUri()).useExpiredToken(true)
+            .audience(configurationProperties.getClientId()).build();
+        DartsTokenAndJwksKey tokenDetails = token.fetchTokenWithGlobalUser();
+
+        mockMvc.perform(get(ENDPOINT_URL).header("Authorization", "Bearer " + tokenDetails.getToken()))
+            .andExpect(status().isUnauthorized())
+            .andReturn();
+    }
+
+    @Test
+    void testNoToken() throws Exception {
+        DartsTokenGenerator token = DartsTokenGenerator.builder().issuer("test")
+            .audience(configurationProperties.getClientId()).build();
+        DartsTokenAndJwksKey tokenDetails = token.fetchTokenWithGlobalUser();
+
+        tokenStub.stubExternalJwksKeys(tokenDetails.getJwksKey());
+
+        mockMvc.perform(get(ENDPOINT_URL).header("Authorization", ""))
+            .andExpect(status().is3xxRedirection())
+            .andExpect(header().exists("Location"))
+            .andReturn();
+    }
+
+    @Test
+    void testInactiveUserCheck() throws Exception {
+        DartsTokenGenerator token = DartsTokenGenerator.builder().issuer(configurationProperties.getIssuerUri())
+            .audience(configurationProperties.getClientId()).build();
+        DartsTokenAndJwksKey tokenDetails = token.fetchTokenWithGlobalUser();
+
+        userAccountStub.setInactiveState("darts.global.user@hmcts.net", false);
+
+        tokenStub.stubExternalJwksKeys(tokenDetails.getJwksKey());
+
+        mockMvc.perform(get(ENDPOINT_URL).header("Authorization", "Bearer " + tokenDetails.getToken()))
+            .andExpect(status().isForbidden())
+            .andExpect(content().json("{\"type\":\"AUTHORISATION_106\",\"title\":\"Could not obtain user details\",\"status\":403}"))
+            .andReturn();
+    }
+
+    @Test
+    void testSuccessfulCall() throws Exception {
+        DartsTokenGenerator token = DartsTokenGenerator.builder().useGlobalKey(true).issuer(configurationProperties.getIssuerUri())
+            .audience(configurationProperties.getClientId()).build();
+        DartsTokenAndJwksKey tokenDetails = token.fetchTokenWithGlobalUser();
+
+        tokenStub.stubExternalJwksKeys(tokenDetails.getJwksKey());
+
+        mockMvc.perform(get(ENDPOINT_URL).header("Authorization", "Bearer " + tokenDetails.getToken()))
+            .andExpect(status().is2xxSuccessful());
+    }
+
+    // TODO: We need to enable audience checking in our security layer
+    @SuppressWarnings({"PMD.DetachedTestCase", "PMD.SignatureDeclareThrowsException"})
+    //@Test
+    void testInvalidAudience() throws Exception {
```

Comment:
We need the darts api to validate against invalid audiences which it does not at the moment. See SecurityConfig

Reply:
Not thought to be a big problem at the time of writing